util/network/cniprovider/createns_linux.go

```diff
@@ -44,16 +58,41 @@ func createNetNS(p string) error {
 		}
 
 		if err != nil {
-			return errors.Wrapf(err, "failed to find pid=%d process", pid)
+			deleteNetNS(p)
+			return "", errors.Wrapf(err, "failed to find pid=%d process", pid)
 		}
 		errno = syscall.Errno(ws.ExitStatus())
 		if errno != 0 {
-			return errors.Wrap(errno, "failed to mount")
+			deleteNetNS(p)
+			return "", errors.Wrap(errno, "failed to mount")
 		}
-		return nil
+		return p, nil
 	}
 	afterForkInChild()
 	_, _, errno = syscall.RawSyscall6(syscall.SYS_MOUNT, uintptr(unsafe.Pointer(procNetNSBytes)), uintptr(unsafe.Pointer(pBytes)), 0, uintptr(unix.MS_BIND), 0, 0)
 	syscall.RawSyscall(syscall.SYS_EXIT, uintptr(errno), 0, 0)
 	panic("unreachable")
 }
+
+func setNetNS(s *specs.Spec, nativeID string) {
```

Comment:
fyi, I updated this in https://github.com/moby/buildkit/pull/1582/commits/94c202bfb9cfaa435e9e005f115ea315fd8cc66a

Reply:
Ah, from #1582. Will that go in soon? Easy enough to rebase on top of it, and I can add the relevant API change in my version in the mean-time.

Reply:
#1582 was merged so this needs rebase now